src/ai/backend/manager/api/scaling_group.py

```diff
@@ -27,6 +33,40 @@
     from .context import RootContext
 
 log = BraceStyleAdapter(logging.getLogger(__name__))
+T = TypeVar('T')
+
+
+class DummyHashObject(Generic[T]):
```

Comment:
The purpose of `DummyHashObject` seems to be for `aiotools.lru_cache` to make a consistent hash value (=cache) for each user group even with different `params` argument in `query_wsproxy_version()` function and to make the dict to be hashable.

I think it would be better to use `dataclasses` or `attrs` to make a hashable attribute set instead of `dict` combined with `DummyHashObject`, in terms of conciseness.